lib/open_mic.rb

```diff
@@ -0,0 +1,36 @@
+class OpenMic
+
+  attr_reader :location, :date, :performers
+
+  def initialize(info)
+    @location = info[:location]
+    @date = info[:date]
+    @performers = []
+  end
+
+  def welcome(user)
+    @performers << user 
+  end
+
+  def repeated_jokes?
+    performers_joke_id = []
+    @performers.each do |performer|
+      performers_joke_id << performer.jokes_id 
+    end
+    (performers_joke_id.flatten.uniq.length != performers_joke_id.flatten.length)
+  end
```

Comment:
Really nice job, getting this to work--it was a bit tricky!

You don't actually need this performers_joke_id or the performer.jokes_id to use the strategy you chose.  You'd still iterate to get an array of all the performers' jokes (the whole joke objects themselves, not just the ids).  Then, you could just compare the uniq length of the list of all jokes to the length of the full list of jokes to see if they're the same. 
```
  all_jokes = @performers.map do |performer|
      performer.jokes
   end
```

all_jokes.flatten.uniq != all_jokes.flatten